test/py/CMakeLists.txt

```diff
@@ -23,6 +23,23 @@
 #####################################################################################
 
 include(PythonModules)
+set(FIX_NAME SETUP_VIRTUAL_ENV)
+set(VENV_DIR ${CMAKE_BINARY_DIR}/test/py/venv)
+
+function(add_py_venv_fixture FIXTURE_NAME VIRTUAL_ENV_DIR REQUIREMENTS_FILE)
+    message("Attempting to create virtual environment: ${VIRTUAL_ENV_DIR}")
+    find_package (Python3 COMPONENTS Interpreter)
+    add_test (NAME INITIALIZE_VENV COMMAND "${Python3_EXECUTABLE}" -m venv ${VIRTUAL_ENV_DIR})
```

Comment:
The name of the test should be `${FIXTURE_NAME}_INITIALIZE_VENV` to avoid conflicts with different python versions or fixtures.